### PR TITLE
Fix network policy

### LINKED ISSUE #669: GitHub events time out with network policy
## Expected Behaviour

OpenFaaS Cloud should function normally when installed using ofc-bootstrap with [network policies enabled](https://github.com/openfaas-incubator/ofc-bootstrap/blob/master/USER_GUIDE.md#toggle-network-policies-recommended).

## Current Behaviour

GitHub events time out with network policies enabled

[`audit-event`](https://github.com/openfaas/openfaas-cloud/blob/master/audit-event/handler.go#L32-L38) logs include:
```
2020/08/25 22:55:20 Duration: 30.010786 seconds
2020/08/25 22:55:20 Post http://gateway.openfaas:8080/function/echo: dial tcp 10.43.99.125:8080: i/o timeout
audit-event: done
```

## Possible Solution

Allow `audit-event` to access `gateway` (since the default [`slack_url`](https://github.com/openfaas/openfaas-cloud/blob/master/slack.yml) is the echo function) either with an explicit policy rule or by giving `audit-event` the [`role: openfaas-system` label](https://github.com/openfaas/openfaas-cloud/blob/master/yaml/network-policy/ns-openfaas-net-policy.yml#L20).

## Steps to Reproduce (for bugs)

1. Install k3s with calico
```
k3sup install --ip $IP --user $USER --k3s-extra-args "--flannel-backend=none --cluster-cidr=192.168.0.0/16 --no-deploy traefik"
kubectl apply -k github.com/wilsonianb/calico-k3s
```
2. Follow ofc-bootstrap [user guide](https://github.com/openfaas-incubator/ofc-bootstrap/blob/master/USER_GUIDE.md) and enable network policy
3. Follow OpenFaaS Cloud [user guide](https://docs.openfaas.com/openfaas-cloud/user-guide/) and attempt to deploy a function
4. Github event will time out

## Context
I'm having to modify network policies myself to get OpenFaaS Cloud to work with network policies enabled.

## Your Environment
* FaaS-CLI version ( Full output from: `faas-cli version` ): `0.12.9`

* Docker version `docker version` (e.g. Docker 17.0.05 ):

* Are you using Docker Swarm or Kubernetes (FaaS-netes)? FaaS-netes on k3s

* Operating System and version (e.g. Linux, Windows, MacOS): CentOS 7

* Link to your project or a code example to reproduce issue:

* Please also follow the [troubleshooting guide](https://github.com/openfaas/faas/blob/master/guide/troubleshooting.md) and paste in any other diagnostic information you have:

### PR DESCRIPTION
**Pull request by wilsonianb**
## Description

Fixes #669 by giving `audit-event` access to `gateway` since echo function is the default `slack_url`.
This would need to be reflected in ofc-bootstrap's [stack.yml](https://github.com/openfaas-incubator/ofc-bootstrap/blob/master/templates/stack.yml) (as should https://github.com/openfaas/openfaas-cloud/commit/2c836f32b44dc354873f72e59f7b0f2d649704bf).

Re-adds support for legacy ingress nginx network policy label (https://github.com/openfaas/openfaas-cloud/commit/1713d3294db4ae308bd51426349acd3e12bc84a4).

## How Has This Been Tested?

Tested on k3s with calico
```
k3sup install --ip $IP --user $USER --k3s-extra-args "--flannel-backend=none --cluster-cidr=192.168.0.0/16 --no-deploy traefik"
kubectl apply -k github.com/wilsonianb/calico-k3s
```
Followed ofc-bootstrap [user guide](https://github.com/openfaas-incubator/ofc-bootstrap/blob/master/USER_GUIDE.md) (without [oauth](https://github.com/openfaas-incubator/ofc-bootstrap/blob/master/USER_GUIDE.md#use-authz-recommended)) with [network policies](https://github.com/openfaas-incubator/ofc-bootstrap/blob/master/USER_GUIDE.md#toggle-network-policies-recommended) enabled and [`openfaas_cloud_version`](https://github.com/openfaas-incubator/ofc-bootstrap/blob/master/example.init.yaml#L334) set to `0.14.1`

Used pre-release ofc-bootstrap [v0.9.6](https://github.com/openfaas-incubator/ofc-bootstrap/tree/0.9.6) with `role: openfaas-system` label added to `audit-event` in [`templates/stack.yml`](https://github.com/openfaas-incubator/ofc-bootstrap/blob/master/templates/stack.yml#L204).

GitHub events no longer time out with this PR change.

Ran helm chart tests with `go test` in `$GO_PATH/src/openfaas-cloud/chart/test/` and ran `diff` on `chart/test/tmp/openfaas-cloud/templates/network-policy/ns-openfaas-net-policy.yml` and `yaml/network-policy/ns-openfaas-net-policy.yml` files.

## How are existing users impacted? What migration steps/scripts do we need?

Should only help OpenFaaS Cloud cluster administrators who had network policies previously enabled and re-run ofc-bootstrap.

## Checklist:

I have:

- [x] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [x] added unit tests

### CODE REVIEW

File: chart/openfaas-cloud/templates/network-policy/ns-openfaas-net-policy.yml [hunk 1]
```diff
@@ -18,7 +18,3 @@ spec:
       podSelector:
         matchLabels:
           role: openfaas-system
-    - namespaceSelector: {}
-      podSelector:
-        matchLabels:
-          app: nginx-ingress
```
Comment (alexellis) [inline chart/openfaas-cloud/templates/network-policy/ns-openfaas-net-policy.yml:24]: Why was this removed? Shouldn't it be ingress-nginx now?

File: yaml/network-policy/edge-router-net-policy.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    - Ingress
+  podSelector:
+    matchLabels:
+      app: edge-router
+  ingress:
+  - from:
+    - namespaceSelector: {}
+      podSelector:
+        matchLabels:
+          app.kubernetes.io/name: ingress-nginx
```
Comment (alexellis) [inline yaml/network-policy/edge-router-net-policy.yml:17]: Is it possible to keep both for legacy users? ingress-nginx and nginx-ingress?

**Review by alexellis — COMMENTED** [review]
Couple of comments

File: yaml/network-policy/edge-router-net-policy.yml (hunk 2 above)
Comment (wilsonianb) [reply yaml/network-policy/edge-router-net-policy.yml:17]: re-added the old label
https://github.com/openfaas/openfaas-cloud/pull/667/commits/95ac2e4ef2dd4b458322d8205f74d5a433c6b085

**Comment by alexellis** [comment]
Please could you raise an issue with the issues you've found?

The Nginx Ingress label name change is here:
https://github.com/openfaas/openfaas-cloud/commit/1713d3294db4ae308bd51426349acd3e12bc84a4

As I mentioned on Slack, I think it will take me a day away from work to verify this PR, so whatever we can do to make that smoother would go a long way.

**Review by Waterdrips — APPROVED** [review]
LGTM

**Comment by alexellis** [comment]
@wilsonianb did you get an issue raised? If so please link it back here through adding a comment or editing its description - GitHub will work out the rest.

**Review by alexellis — APPROVED** [review]
Approved

**Comment by alexellis** [comment]
There was a much bigger PR I remembered, was that not needed? Can you link it here if it is still around?

**Comment by wilsonianb** [comment]
here's the original changeset
https://github.com/openfaas/openfaas-cloud/compare/master...wilsonianb:og-net-policy
This ended up being much smaller since:
- I realized metrics already had the `openfaas-system` role (It just isn't reflected yet in ofc-bootstrap's stack.yml)
- I opted to stick with the `openfaas-system` role model, instead of adding additional network policy rules (which would help constrain pods to only access exactly what they need, but would likely be harder to maintain)
